File: p14.m
% Learning Point detector

close all
clear all
clc

pkg load image

a=double(rgb2gray(imread('img2.bmp')));
figure
imshow(uint8(a));
title('Original Image','fontsize',20)

[r c]=size(a);
b=zeros(r,c);

% Filters
filter1=[-1 -1 -1; ...
         -1  8 -1; ...
         -1 -1 -1;];

b=imfilter(a,filter1); 

b=b-min(min(b));
b=255*(b/max(max(b)));
figure
imshow(uint8(b));
title('Filtered NORMALISED Image','fontsize',20);

File: p14.py
# Learning Point detector
import numpy as np
import cv2
import matplotlib.pyplot as plt

a=np.float32(cv2.imread('./wb.jpg', 0))

cv2.imshow("Original Image", np.uint8(a))


# Filters
filter1=np.array([
                  [-1,-1,-1],
                  [-1,8,-1],
                  [-1,-1,-1]
                ])

b=cv2.filter2D(a,-1,filter1)
b=b-b.min()
b=255*(b/b.max())
cv2.imshow("Filtered Image", np.uint8(b))

cv2.waitKey(0)
cv2.destroyAllWindows()